ui/src/app/machines/views/MachineDetails/MachineTests/MachineTestsTable/MachineTestsTable.tsx

```diff
@@ -21,31 +27,109 @@ const isSuppressible = (result: ScriptResult) =>
   result.status === scriptStatus.TIMEDOUT ||
   result.status === scriptStatus.FAILED_APPLYING_NETCONF;
 
+const renderExpandedContent = (
+  result: ScriptResult,
+  history: ScriptResultHistory,
+  hasVisibleHistory: boolean,
+  hasVisibleMetrics: boolean
+) => {
+  return (
+    <div>
+      {hasVisibleHistory ? (
+        <table role="grid" className="p-table-expanding--light">
+          <tbody>
+            {history[result.id]?.map((item: PartialScriptResult) => {
+              if (item.id !== result.id) {
+                return (
+                  <tr
+                    role="row"
+                    data-test="script-result-history"
+                    key={`history-${item.id}`}
+                  >
+                    <td role="gridcell"></td>
+                    <td role="gridcell"></td>
+                    <td role="gridcell"></td>
+                    <td role="gridcell">{item.runtime}</td>
+                    <td role="gridcell">{item.updated}</td>
+                    <td role="gridcell">{item.status_name}</td>
+                    <td role="gridcell"></td>
+                  </tr>
+                );
+              } else {
+                return null;
+              }
+            })}
+          </tbody>
+        </table>
+      ) : null}
+      {hasVisibleMetrics ? (
+        <table role="grid" className="p-table-expanding--light">
+          <tbody>
+            {result.results.map((item: ScriptResultResult) => (
+              <tr
+                role="row"
+                data-test="script-result-metrics"
+                key={`metric-${item.name}`}
+              >
+                <td role="gridcell">{item.title}</td>
```

Comment:
This reminds me we have a component `LabelledList` which we've used to display key-value pairs in other places, e.g.
![2021-01-27_11-26](https://user-images.githubusercontent.com/25733845/105929104-9bc80200-6092-11eb-8f64-78ea8491ec77.png)

Not saying it needs to go in this PR, just something to consider I guess.